keras_cv/layers/efficient_multihead_attention.py

```diff
@@ -0,0 +1,116 @@
+import math
+
+from keras_cv.backend import keras
+from keras_cv.backend import ops
+
+
+@keras.saving.register_keras_serializable(package="keras_cv")
+class SegFormerMultiheadAttention(keras.layers.Layer):
+    def __init__(self, project_dim, num_heads, sr_ratio):
+        """
+        Efficient MultiHeadAttention implementation as a Keras layer.
+        A huge bottleneck in scaling transformers is the self-attention layer with an O(n^2) complexity.
+
+        SegFormerMultiheadAttention performs a sequence reduction (SR) operation with a given ratio, to reduce
+        the sequence length before performing key and value projections, reducing the O(n^2) complexity to O(n^2/R) where
```

Comment:
Is this not (n/R)^2

Reply:
Took this from the paper, so I honestly took it with blind faith

Reply:
![image](https://github.com/keras-team/keras-cv/assets/60978046/cda536e8-dad0-4f62-b8e5-19263bdb8d4a)